src/odfesqlodbc/results.c

```diff
@@ -1388,6 +1390,17 @@ RETCODE SQL_API ESAPI_ExtendedFetch(HSTMT hstmt, SQLUSMALLINT fFetchType,
      */
     if (!should_set_rowset_start)
         rowset_start = SC_get_rowset_start(stmt);
+
+    // Get more results when cursor reaches end 
+    if ((!useCursor) && (rowset_start >= num_tuples) && (res->server_cursor_id != NULL)) {
+        conn = SC_get_conn(stmt);
+        if (conn != NULL) {
+            ESSendCursorQueries(conn->esconn, res->server_cursor_id);
+            GetNextResultSet(stmt);
+            num_tuples = QR_get_num_total_tuples(res);
```

Comment:
Do you need to reset rowset_start?

Reply:
Not here since the result at this index isn't returned yet